lavague-core/lavague/core/extractors.py

```diff
@@ -16,19 +17,32 @@ class JsonFromMarkdownExtractor(BaseExtractor):
     Completion:
     --------------------------------------------
     """
-
-    def extract(self, markdown_text: str) -> str:
+   
+    def extract(self, markdown_text: str, action_shape_validator = None) -> str:
```

Comment:
Since the extractor can be used with structures beyond just actions, I suggest naming the validator `shape_validator`